### DISCUSSION TITLE
How to represent a Rule which returns a value, in a string in another expression

### DISCUSSION THREAD

**Discussion by arrafeeahsan** [microsoft/RulesEngine#436] [Q&A]
{
  "Name": "RemainingCount",
  "Expression": "ActiveMemberInGroup - DefaultActiveMemberNumberPerGroup"
},
{
  "Name": "Message",
  "Expression": "RemainingCount > 0 ?  \"Maximum number are **{RemainingCount}** \" :  \"Maximum Number are 0!\" "
}

**Comment by bmasotta-lcg** [comment]
You can use an OnSuccess Action of the type (Name) OutputExpression and then specify your expression in the context Expression
You will be able to see the result in the RuleResultTree.ActionResult field 

 ```
  "Actions": {
          "OnSuccess": {
            "Name": "OutputExpression",
            "Context": {
              "Expression": "CustomType.GenerateMessage(GlobalParamSomething, RuleParamOther)"
            }
          },
          "OnFailure": {
          }
}
```
